object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -279,6 +304,313 @@ private static String getExactStringValue(Tree tree, ValueAnnotatedTypeFactory f
+
+    if (!visited.stream()
+        .anyMatch(pair -> pair.block.equals(state.block) && pair.localSet.equals(state.localSet))) {
+      visited.add(state);
+      worklist.add(state);
+    }
+  }
+
+  /**
+   * Returns the String value of @AlwaysCall annotation declared on the class type of {@code
+   * element}. Returns null if the class type of {@code element} doesn't have @AlwaysCall
+   * annotation.
+   */
+  private @Nullable String getAlwaysCallValue(Element element) {
+
+    TypeMirror type = element.asType();
+    TypeElement eType = TypesUtils.getTypeElement(type);
+    AnnotationMirror alwaysCallAnnotation = getDeclAnnotation(eType, AlwaysCall.class);
+
+    return (alwaysCallAnnotation != null)
+        ? AnnotationUtils.getElementValue(alwaysCallAnnotation, "value", String.class, false)
+        : null;
+  }
+
+  /**
+   * Creates the appropriate @CalledMethods annotation that corresponds to the @AlwaysCall
+   * annotation declared on the class type of {@code assign.first}. Then, it gets @CalledMethod
+   * annotation of {@code assign.first} to do a subtyping check and reports an error if the check
+   * fails.
+   */
+  private void checkAlwaysCall(
+      Pair<LocalVariableNode, Tree> assign,
+      CFStore store,
+      AnnotatedTypeMirror annotatedTypeMirror) {
+
+    CFValue lhsCFValue = store.getValue(assign.first);
+    String alwaysCallValue = getAlwaysCallValue(assign.first.getElement());
+
+    boolean report = true;
+
+    if (lhsCFValue != null) { // When store contains the lhs
+      AnnotationMirror dummyCMAnno = createCalledMethods(alwaysCallValue);
+      AnnotatedTypeMirror annoType = getAnnotatedType(assign.first.getTree());
+      AnnotationMirror CMAnno = annoType.getAnnotationInHierarchy(TOP);
+      QualifierHierarchy qualifierHierarchy =
+          createQualifierHierarchy(new MultiGraphQualifierHierarchy.MultiGraphFactory(this));
+
+      if (qualifierHierarchy.isSubtype(CMAnno, dummyCMAnno)) {
+        report = false;
+      }
+
+    } else if (annotatedTypeMirror != null) {
+
+      // Sometimes getStoreAfter doesn't contain correct set of local variable nodes! Then, it
+      // checks the AlwaysCall condition by looking at AnnotatedTypeMirror
+      AnnotationMirror annotationMirror = annotatedTypeMirror.getAnnotation(CalledMethods.class);
+      if (annotationMirror != null
+          && getValueOfAnnotationWithStringArgument(annotationMirror).contains(alwaysCallValue)) {
+        report = false;
+      }
+    }
+
+    if (report) {
+      checker.reportError(assign.second, "missing.alwayscall", alwaysCallValue);
+    }
+  }
+
+  private boolean hasAlwaysCall(TypeMirror type) {
+    TypeElement eType = TypesUtils.getTypeElement(type);
+    return ((eType != null) && (getDeclAnnotation(eType, AlwaysCall.class) != null));
+  }
+
+  private class BlockWithLocals {
+    public BlockImpl block;
+    public Set<Pair<LocalVariableNode, Tree>> localSet;
```

Comment:
We might also consider giving `Pair<LocalVariableNode, Tree>` a name - that structure is used extensively above, and also here.